common/expression/src/function.rs

```diff
@@ -698,70 +590,29 @@ pub fn vectorize_passthrough_nullable_2_arg<I1: ArgType, I2: ArgType, O: ArgType
 + Copy {
     move |arg1, arg2, generics| match (arg1, arg2) {
         (ValueRef::Scalar(None), _) | (_, ValueRef::Scalar(None)) => Ok(Value::Scalar(None)),
-        (ValueRef::Scalar(Some(arg1)), ValueRef::Scalar(Some(arg2))) => {
-            Ok(Value::Scalar(Some(func(arg1, arg2))))
-        }
+        (ValueRef::Scalar(Some(arg1)), ValueRef::Scalar(Some(arg2))) => Ok(Value::Scalar(Some(
+            func(ValueRef::Scalar(arg1), ValueRef::Scalar(arg2), generics)?
+                .into_scalar()
+                .unwrap(),
+        ))),
         (ValueRef::Scalar(Some(arg1)), ValueRef::Column((arg2, arg2_validity))) => {
-            let iter = I2::iter_column(&arg2).map(|arg2| func(arg1.clone(), arg2));
-            let col = O::column_from_iter(iter, generics);
+            let col = func(ValueRef::Scalar(arg1), ValueRef::Column(arg2), generics)?
+                .into_column()
+                .unwrap();
             Ok(Value::Column((col, arg2_validity)))
         }
         (ValueRef::Column((arg1, arg1_validity)), ValueRef::Scalar(Some(arg2))) => {
-            let iter = I1::iter_column(&arg1).map(|arg1| func(arg1, arg2.clone()));
-            let col = O::column_from_iter(iter, generics);
+            let col = func(ValueRef::Column(arg1), ValueRef::Scalar(arg2), generics)?
+                .into_column()
+                .unwrap();
             Ok(Value::Column((col, arg1_validity)))
         }
         (ValueRef::Column((arg1, arg1_validity)), ValueRef::Column((arg2, arg2_validity))) => {
-            let iter = I1::iter_column(&arg1)
-                .zip(I2::iter_column(&arg2))
-                .map(|(arg1, arg2)| func(arg1, arg2));
-            let col = O::column_from_iter(iter, generics);
+            let col = func(ValueRef::Column(arg1), ValueRef::Column(arg2), generics)?
+                .into_column()
```

Comment:
If we always unwrap the value returned by into_column, how about expect the error inside into_column instead?

Reply:
but into_columns is derived by EnumAsInner 🤔